Civi/Api4/Generic/Action/Basic/Get.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Civi\Api4\Generic\Action\Basic;
+
+use Civi\Api4\Generic\Action\AbstractGet;
+use Civi\Api4\Generic\Result;
+
+/**
+ * Retrieve items based on criteria specified in the 'where' param.
+ *
+ * Use the 'select' param to determine which fields are returned, defaults to *.
+ */
+class Get extends AbstractGet {
+  use \Civi\Api4\Generic\Action\Traits\ArrayQueryTrait;
+
+  /**
+   * @var callable
+   */
+  private $getter;
+
+  public function __construct($entity, $getter = NULL) {
+    parent::__construct($entity);
+    if ($getter) {
+      $this->getter = $getter;
+    }
+    else {
+      $this->getter = [$this, 'getObjects'];
```

Comment:
This took me a moment to understand -- so is the idea that you can either (1) pass the getter as a callback or (2) extend the class and implement the `getObjects()` function?

If so, I think this would be clearer: https://gist.github.com/totten/f10ceeb1bfe76fc2d6b9844f35a1b688 -- it exposes `getObjects()` as a formal part of the class hierarchy, making it it more obvious that one override it... and allowing a docblock for it.

Reply:
Potential nitpick: the term `getObjects()` might technically suggest... `object`s. A name like `getRecords()`, `getRows()`, `getCandidates()`, or `getData()` might be more appropriate for a list of associative-arrays.

Reply:
@totten I like it - that's much better. Will roll that suggestion in shortly.

One thing I'm learning through this process is that "less is more" when it comes to using magic in php.